doc/declarative-state-allocation/s3.md

```diff
@@ -0,0 +1,78 @@
+Allocate s3 buckets using OAM definition
+===============
+
+This document covers the capability to declaratively allocate state using AWS [S3](https://aws.amazon.com//s3/) (Simple Storage Service) buckets. This capability uses [CrossPlane](https://crossplane.io/) addon to interact with AWS APIs and create cloud resources using straightforward Application definitions from OAM/kubeVela.
+
+
+Creating a bucket
+---------------
+
+To create a new bucket we need to set up an Application with this component definition:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: oam-bucket-example
+  namespace: default
+spec:
+  components:
+  - name: oam-bucket-example
+    type: bucket
+```
+
+This Application will create some CrossPlane objects on top of Kubernetes:
+
+- Bucket
+- Role
+- Policy
+- RolePolicyAttachment
+
+It is important to note that s3 bucket names are globally unique. If we try to create an OAM Application with s3 component using a name of an existing bucket, we have these behaviors:
+
+1. If the target bucket is from our AWS account:<br>
+    CrossPlane will import the target and start managing its state. This is exactly the same as importing some living resource state using Terraform.
+
+2. If the target bucket is from another AWS account:<br>
+    CrossPlane will create all the necessary CRDS in the cluster but they will never reach the Synced/Ready state, and then, the bucket will never be available.
+
+**⚠️ IMPORTANT**!
+
+The s3 bucket is tied to the parent Application lifecycle so if the Application is deleted (or just recycled through VelaUX, for example), the bucket is deleted as well (if empty). If the bucket is not empty, it's necessary to remove all the objects stored there before deleting it. 
```

Comment:
We should track some followup work to have bucket garbage collection include deleting the bucket objects automatically.